test/e2e-common.sh

```diff
@@ -210,6 +211,85 @@ function enable_monitoring(){
       --role roles/cloudtrace.agent
 }
 
+# test_authentication_check_for_brokercell tests the authentication check function for BrokerCell.
+# This test required the entire BrokerCell to be non-functional.
+# In order to avoid running in parallel with other tests, we put it in the shell script, rather than more common go tests.
+# TODO Once we support multiple BrokerCells with distinct authentication mechanisms, make this a normal test that runs in parallel with the others.
```

Comment:
What's the plan to run this test now? A new Prow job?

Reply:
This test will only run Integration test workflow in secret mode and integration test workflow in wi mode. Since we reuse the test_setup for upgrade test and conformance test, an extra-env `ENABLE_AUTH_CHECK_TEST` is added into knative/test-infra#2637 to differentiate whether a test workflow needs auth check test or not.

Reply:
Make sense. From the change in test-infra, this will only run in integration test and workload identity. lgtm